docs/knn/index.md

```diff
@@ -246,3 +246,98 @@ All parameters are required.
 The standard KNN query and custom scoring option perform differently. Test using a representative set of documents to see if the search results and latencies match your expectations.
 
 Custom scoring works best if the initial filter reduces the number of documents to no more than 20,000. Increasing shard count can improve latencies, but be sure to keep shard size within [the recommended guidelines](../elasticsearch/#primary-and-replica-shards).
+
+
+## Use Similarity methods in painless scripting
+
+Sometimes users would like to go beyond Elasticsearch’s built-in features for scoring and might want to customize the search scores in more complex ways.
```

Comment:
This whole section needs some general language tune-up and can probably lose a bit of content. We frame everything as "you" rather than "users" or "customers," so something like:

If you want to go beyond Elasticsearch's built-in scoring features, you can use KNN similarity functions in Painless scripts. Painless has a strict list of allowed functions and classes per context to ensure its scripts are secure. The KNN plugin lets you use the following functions within the score context.

(Of course add/tweak anything if I've lost meaning.)